pkg/controllers/dashboard/helm/repo_oci.go

```diff
@@ -56,18 +73,37 @@ func (o *OCIRepohandler) onClusterRepoChange(key string, clusterRepo *catalog.Cl
 	if clusterRepo == nil {
 		return nil, nil
 	}
-
 	// Ignore non OCI ClusterRepos
 	if !registry.IsOCI(clusterRepo.Spec.URL) {
 		return clusterRepo, nil
 	}
 
+	updatedRepo, err := o.clusterRepoController.Get(key, metav1.GetOptions{})
+	if err == nil && updatedRepo.ResourceVersion != clusterRepo.ResourceVersion {
```

Comment:
What is there is an error ?

Reply:
If there is an error, we just move forward because this is a check to prevent calling the registry when the controller was triggered for an outdated version of the CRD